integration-test-groups/aws2-quarkus-client/aws2-s3/src/main/java/org/apache/camel/quarkus/component/aws2/s3/it/Aws2S3QuarkusClientInvoker.java

```diff
@@ -16,9 +16,15 @@
  */
 package org.apache.camel.quarkus.component.aws2.s3.it;
 
-import io.quarkus.test.junit.NativeImageTest;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
 
-@NativeImageTest
-class Aws2S3QuarkusClientIT extends Aws2S3QuarkusClientTest {
+import software.amazon.awssdk.services.s3.S3Client;
+
+@ApplicationScoped
+public class Aws2S3QuarkusClientInvoker {
+
+    @Inject
+    S3Client s3;
```

Comment:
So, we inject the S3Client here. But the class is not @Unremovable as stated in the doc https://camel.apache.org/camel-quarkus/2.6.x/reference/extensions/aws2-s3.html#_optional_integration_with_quarkus_amazon_s3.

What do you think ?

Reply:
To be honest I'm not sure. My guess is that in some cases, quarkus my optimize code and remove unused injections. But our tests does not allow it. So it works without it. (but it may be required with different configuration)

Reply:
Ok, Quarkus might not detect that this bean is needed to be removed in our test configuration then. Thanks for sharing.